src/QMCWaveFunctions/Fermion/BackflowTransformation.h

```diff
@@ -156,15 +156,17 @@ class BackflowTransformation : public OptimizableObject
 
   void restore(int iat = 0);
 
-  void checkInVariables(opt_variables_type& active) override;
-
-  void reportStatus(std::ostream& os) override;
-
-  void checkOutVariables(const opt_variables_type& active) override;
-
   bool isOptimizable() const;
-
-  void resetParameters(const opt_variables_type& active) override;
+  void checkInVariables(opt_variables_type& active);
+  void checkOutVariables(const opt_variables_type& active);
+  void resetParameters(const opt_variables_type& active);
+  void reportStatus(std::ostream& os) final;
+
+  // extractOptimizableObjectRefs is not enabled in BackflowTransformation.
```

Comment:
I assume this is part of why this object is final?  Not sure whether this needs to be documented more at this time but it leaves me wondering why BackFlowTransformation is a special case.  This probably really amounts to BackFlow needing better documentation which is beyond this PR.

Reply:
Right now there is a compiler warning. Adding final in the class stops the warning. BackFlowTransformation should never be inherited and thus I think adding final is OK.